spec/controllers/builds_controller_spec.rb

```diff
@@ -139,6 +139,42 @@
     end
   end
 
+  describe "#on_success_log_file link" do
+    render_views
+    let(:build) { FactoryGirl.create(:build, state: :succeeded) }
+    before do
+      @action = :show
+      @params = {:id => build.id, :repository_path => build.repository}
+    end
+
+    context "has on_success_log_file" do
+      before do
+        output = "Exited with status: 0"
+        script_log = FilelessIO.new(output)
+        script_log.original_filename = "on_success_script.log"
+        build.on_success_script_log_file = script_log
+        build.save
+      end
+
+      it "displays link to on_success_log_file" do
+        get @action, @params
+        doc = Nokogiri::HTML(response.body)
+        elements = doc.css("#content > div > a")
+        expect(elements.size).to eq(1)
+        expect(elements[0].content).to eq("on_success_script.log")
+      end
+    end
+
+    context "does not have on_success_log_file" do
+      it "does not display link to on_success_log_file" do
+        get @action, @params
+        doc = Nokogiri::HTML(response.body)
+        elements = doc.css("#content > div > a")
```

Comment:
Should another link gets added to this page this test will fail. What about searching for the string "on_success_script" because that's going to be the link text.